java-squid/src/main/java/org/sonar/java/se/NotEqualRelation.java

```diff
@@ -36,25 +43,35 @@ protected String getOperand() {
   }
 
   @Override
-  protected Boolean implies(SymbolicValueRelation relation) {
-    if (hasSameOperand(relation)) {
-      return relation instanceof NotEqualRelation;
-    }
-    return null;
+  protected Boolean isImpliedBySymbolicValueRelation(SymbolicValueRelation relation) {
+    return relation.impliesNotEqual();
   }
 
   @Override
-  protected SymbolicValueRelation underHypothesis(SymbolicValueRelation relation) {
-    return relation.hypothesisWithNotEqual(this);
+  protected Boolean impliesEqual() {
+    return Boolean.FALSE;
   }
 
   @Override
-  protected SymbolicValueRelation hypothesisWithEqual(EqualRelation relation) {
-    return null;
+  protected Boolean impliesNotEqual() {
+    return Boolean.TRUE;
+  }
+
+  @Override
+  @CheckForNull
+  protected SymbolicValueRelation combinedAfterSymbolicValueRelation(SymbolicValueRelation relation) {
+    return relation.combinedWithNotEqual(this);
+  }
+
+  @Override
+  @CheckForNull
+  protected SymbolicValueRelation combinedWithEqual(EqualRelation relation) {
+    return new NotEqualRelation(v1, relation.v2);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Null is returned but a "Boolean" is expected. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2447)